db_dtypes/__init__.py

```diff
@@ -52,6 +56,26 @@ class TimeDtype(core.BaseDatetimeDtype):
     def construct_array_type(self):
         return TimeArray
 
+    @staticmethod
+    def __from_arrow__(
+        array: Union[pyarrow.Array, pyarrow.ChunkedArray]
+    ) -> "TimeArray":
+        # We can't call combine_chunks on an empty array, so short-circuit the
+        # rest of the function logic for this special case.
+        if len(array) == 0:
+            return TimeArray(numpy.array([], dtype="datetime64[ns]"))
+
+        # We can't cast to timestamp("ns"), but time64("ns") has the same
+        # memory layout: 64-bit integers representing the number of nanoseconds
+        # since the datetime epoch (midnight 1970-01-01).
+        array = pyarrow.compute.cast(array, pyarrow.time64("ns"))
+
+        # ChunkedArray has no "view" method, so combine into an Array.
+        array = array.combine_chunks() if hasattr(array, "combine_chunks") else array
```

Comment:
Would the intent be clearer here with an `isinstance` test?  E.g.:
```suggestion
        if isinstance(array, ChunkedArray):
            array = array.combine_chunks()
```

Reply:
Done. That'll be clearer for mypy to figure out once we add type checking here, too.